components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/threatprotection/analyzer/JSONAnalyzer.java

```diff
@@ -0,0 +1,196 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * Implementation of APIMThreatAnalyzer for JSON Payloads
+ */
+public class JSONAnalyzer implements APIMThreatAnalyzer {
+
+    private static final String JSON_THREAT_PROTECTION_MSG_PREFIX = "Threat Protection-JSON: ";
+    private JsonFactory factory;
+    private boolean enabled = true;
+    private int maxFieldCount = 0;
+    private int maxStringLength = 0;
+    private int maxArrayElementCount = 0;
+    private int maxFieldLength = 0;
+    private int maxJsonDepth = 0;
+
+    public JSONAnalyzer() {
+        factory = new JsonFactory();
+    }
+
+    /**
+     * Create a JSONAnalyzer using default configuration values
+     */
+    public void configure(JSONConfig config) {
+        enabled = config.isEnabled();
+        maxFieldCount = config.getMaxPropertyCount();
+        maxStringLength = config.getMaxStringLength();
+        maxArrayElementCount = config.getMaxArrayElementCount();
+        maxFieldLength = config.getMaxKeyLength();
+        maxJsonDepth = config.getMaxJsonDepth();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public void configure(XMLConfig config) {
+        throw new UnsupportedOperationException("This method is not supported on this instance");
+    }
+
+    /**
+     * @param in payload
+     * @throws APIMThreatAnalyzerException if defined limits for json payload exceeds
+     */
+    @Override
+    public void analyze(InputStream in, String apiContext) throws APIMThreatAnalyzerException {
+        try (JsonParser parser = factory.createParser(new InputStreamReader(in))) {
+            int currentDepth = 0;
+            int currentFieldCount = 0;
+            JsonToken token;
+            while ((token = parser.nextToken()) != null) {
+                switch (token) {
+                    case START_OBJECT:
+                        currentDepth += 1;
+                        analyzeDepth(maxJsonDepth, currentDepth, apiContext);
+                        break;
+
+                    case END_OBJECT:
+                        currentDepth -= 1;
+                        break;
+
+                    case FIELD_NAME:
+                        currentFieldCount += 1;
+                        String name = parser.getCurrentName();
+                        analyzeField(name, maxFieldCount, currentFieldCount, maxFieldLength);
+                        break;
+
+                    case VALUE_STRING:
+                        String value = parser.getText();
+                        analyzeString(value, maxStringLength);
+                        break;
+
+                    case START_ARRAY:
+                        analyzeArray(parser, maxArrayElementCount, maxStringLength);
+                }
+            }
+        }
+        catch (JsonParseException e) {
+            throw new APIMThreatAnalyzerException("Error occurred while parsing the JSON payload", e);
```

Comment:
Will it be useful to include api context with the error message

Reply:
in code review, we discuss only put generic messages.